tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Signature.java

```diff
@@ -32,6 +34,16 @@
   /** The default signature key, when not provided */
   public static final String DEFAULT_KEY = "serving_default";
 
+  public static class TensorDescription {
+    public final DataType dataType;
+    public final long[] shape;
```

Comment:
Hi @jbentleyEG , looks good but as we discussed, can we return the shape as a `Shape` object to the users instead of a `long[]`?

Reply:
Hi @jbentleyEG, just checking, do you agree to make the proposed change above?

Reply:
Sorry for the flakiness, I keep getting pulled away by other tasks. I wasn't able to generate the Shape directly from the TensorShapeProto, but I was able to recreate it from the dimensions.